dogfood.md

```diff
@@ -0,0 +1,167 @@
+
+### Cordova SDK
+GraphQL seems to have a lot of javascript support which will be useful for our Javascript/Cordova client application.  The Cordova SDK has all of the features we will use, and we have a lot of JavaScript expertise to draw on.
+
+### iOS
+
+The iOS SDK has all of the features we need, and iOS has an Apollo GraphQL library as well.  Like Cordova and Android there is a lot of experience we can draw on for developing the app.
+
+
+## Data Model and API layer
+
+The following is a graphql schema for the proposed data model.  The GraphQL schema also includes the public APIs that will be implemented.
+
+### Sample Schema
+
+```graphql
+type Profile {
+  id: ID! @unique
+  email: String!
+  display_name: String!
+  biography: String!
+  pictureUrl: String!
+}
+
+type MemeList {
+  id: ID! @unique
+  ownerId: ID!
+  createdOn: String!
+  updatedOn: String!
+  memes: [Meme!]!
+}
+
+type Meme {
+  id: ID! @unique
+  ownerId: ID!
+  createdOn: String!
+  updatedOn: String!
+  photoUrl: String!
+  comments: [Comment!]!
+}
+
+type Comment {
+  id: ID! @unique
+  ownerId: ID!
+  createdOn: String!
+  updatedOn: String!
+  comment: String!
+}
+
+type Query {
+  profile(email: String!):Profile!
+  usersMemes(email: String!): [Meme!]!
+  feed(memeListId: ID!): [Meme!]!
+  meme(memeId: ID!):Meme
+}
+
+type Mutation {
+  createProfile(email: String!, display_name: String!, biography: String!, pictureUrl: String!):Profile!
+  updateProfile(id: ID!, email: String!, display_name: String!, biography: String!, pictureUrl: String!):Profile!
+  deleteProfile(id: ID!):Profile!
+
+  postMeme(picture: [Int!]!, topComment: String!,bottomComment: String!, memeListId: ID!): Meme!
+  deleteMeme(id: ID!): Meme!
+  likeMeme(id: ID!): Meme!
+  unlikeMeme(id: ID!): Meme!
+
+  postComment(memeId: ID!, comment: String!): Meme!
+  deleteComment(id: ID!): Meme!
+}
+```
+
+## Mobile-Core features used
+
+Memeolist will be build using technologies provided with the mobile-core service catalog.  Additional services will be created as necessary using RHOAR runtimes and exposed via a custom connector.  Finally, we will use this as an opportunity to test graphql style APIs which will serve as the basis for the future sync project.
+
+### Custom Service Connector
+
+The final meme images will be generated server side using a RHOAR runtime (vert.x, node.js, or Wildfly Swarm) and exposed using the custom service connector feature.  This means that our SDKs will be responsible for upload the image, the author metadata, and the message to be displayed on it and the RHOAR service will compose that data into a Meme post and save it to a persistent data store.
+
+### Keycloak
+
+Social integration will be provided with KeyCloak and we will use our auth SDKs.  We should also provide a templated realm and a script so that community users can plug in their developer credentials if they wish to host their own version of memeolist.
```

Comment:
@darahayes @aliok there's a card on the backlog for authentication
https://trello.com/c/iaLvkeB8/87-13-as-a-developer-i-want-to-be-able-to-protect-my-data-sync-service-with-keycloak

and authorisation
https://trello.com/c/15PCmsz3/9-as-a-security-conscious-developer-i-want-to-be-able-to-apply-role-permission-checks-on-incoming-sync-requests-so-that-i-can-ensu

We'll need to consider authentication after the initial end to end with the memeolist app and data sync.
Authorization will be considered at some point after that

Reply:
@david-martin you got me a little confused here. I think the idea is use Auth SDK to the user authenticateuse the app, so I think we will need to considered it **before** the initial end to end with the memeolist app

Reply:
The app could certainly have authentication (using auth sdk).
However, the authentication bit on the data sync server (integration with keycloak for checking token validitly) may not be there with the first end to end